Entities/AtomGroup.cpp

```diff
@@ -1,2618 +1,1750 @@
+
+				// Get the MO we seem to be intersecting
+				MovableObject *tempMO = g_MovableMan.GetMOFromID(hitMOID);
+				RTEAssert(tempMO, "Intersected MOID couldn't be translated to a real MO!");
+				tempMO = tempMO->GetRootParent();
+
+				if (tempMO->GetsHitByMOs()) {
+					// Make that MO draw itself again in the MOID layer so we can find its true edges
+					intersectedMO = tempMO;
+					intersectedMO->Draw(g_SceneMan.GetMOIDBitmap(), Vector(), g_DrawMOID, true);
+					break;
+				}
+			}
+		}
+		if (!intersectedMO) {
+			return false;
+		}
+
+		// Tell both MO's that they have hit an MO, and see if they want to continue
+		if (m_OwnerMO->OnMOHit(intersectedMO)) {
+			return false;
+		}
+		if (intersectedMO->OnMOHit(m_OwnerMO->GetRootParent())) {
+			return false;
+		}
+
+		std::list<Atom *> intersectingAtoms;
+
+		// Restart and go through all Atoms to find all intersecting the specific intersected MO
+		for (Atom *atom : m_Atoms) {
+			atomPos = atom->GetCurrentPos();
+			if (g_SceneMan.GetMOIDPixel(atomPos.GetFloorIntX(), atomPos.GetFloorIntY()) == currentMOID) {
+				// Add atom to list of intersecting ones
+				intersectingAtoms.push_back(atom);
+			}
+		}
+
+		Vector exitDirection = Vector();
+		Vector atomExitVector = Vector();
+		Vector totalExitVector = Vector();
+
+		// Go through all intersecting Atoms and find their average inverse normal
+		for (const Atom *intersectingAtom : intersectingAtoms) {
+			exitDirection += m_OwnerMO->RotateOffset(intersectingAtom->GetNormal());
+		}
+
+		// We don't have a direction to go, so quit
+		// TODO: Maybe use previous position to create an exit direction instead then?
+		if (exitDirection.IsZero()) {
+			return false;
+		}
+
+		// Invert and set appropriate length
+		exitDirection = -exitDirection;
+		exitDirection.SetMagnitude(m_OwnerMO->GetDiameter());
+
+		Vector clearPos = Vector();
+
+		// See which of the intersecting Atoms has the longest to travel along the exit direction before it clears
+		float longestDistance = 0.0F;
+		for (const Atom *intersectingAtom : intersectingAtoms) {
+			atomPos = intersectingAtom->GetCurrentPos();
+			if (g_SceneMan.CastFindMORay(atomPos, exitDirection, g_NoMOID, clearPos, 0, true, 0)) {
+				// Determine the longest clearing distance so far
+				atomExitVector = clearPos - atomPos.GetFloored();
+				if (atomExitVector.GetMagnitude() > longestDistance) {
+					// We found the Atom with the longest to travel along the exit direction to clear, so that's the distance to move the whole object to clear all its Atoms.
+					longestDistance = atomExitVector.GetMagnitude();
+					totalExitVector = atomExitVector;
+				}
+			}
+		}
+
+		// The final exit movement vectors for this' Owner MO and the MO intersected by this
+		Vector thisExit;
+		Vector intersectedExit;
+
+		// If the other object is pinned, then only move this. Otherwise, apply the object exit vector to both this owner object, and the one it intersected, proportional to their masses.
```

Comment:
Private method.
Also, the comments about only moving some stuff should be consolidated into an explanation of why we're doing things and then probably used as a method description. Something like
`When deciding whether to move objects, we should ignore the other object if it's pinned or  much higher mass than this one.`

Reply:
I'm keeping most of the comments as they are because most of this code is nightmare garbage and you'll thank me one day.

Reply:
Yeah fair enough. They should still be consolidated, but that requires a lot of time and effort figuring out what's going on, so it can be done later.